Source/Core/DolphinQt2/MainWindow.cpp

```diff
@@ -819,3 +833,52 @@ QSize MainWindow::sizeHint() const
 {
   return QSize(800, 600);
 }
+
+void MainWindow::OnBootGameCubeIPL(DiscIO::Region region)
+{
+  StartGame(std::make_unique<BootParameters>(BootParameters::IPL{region}));
+}
+
+void MainWindow::OnImportNANDBackup()
+{
+  auto response = QMessageBox::question(
+      this, tr("Question"),
+      tr("Merging a new NAND over your currently selected NAND will overwrite any channels "
+         "and savegames that already exist. This process is not reversible, so it is "
+         "recommended that you keep backups of both NANDs. Are you sure you want to "
+         "continue?"));
+
+  if (response == QMessageBox::No)
+    return;
+
+  QString file = QFileDialog::getOpenFileName(this, tr("Select the save file"), QDir::currentPath(),
+                                              tr("BootMii NAND backup file (*.bin);;"
+                                                 "All Files (*)"));
+
+  if (file.isEmpty())
+    return;
+
+  QProgressDialog* dialog = new QProgressDialog(this);
+  dialog->setMinimum(0);
+  dialog->setMaximum(0);
+  dialog->setLabelText(tr("Importing NAND backup"));
+  dialog->setCancelButton(nullptr);
+  dialog->show();
+
+  auto beginning = QDateTime::currentDateTime().toSecsSinceEpoch();
+
+  std::future<void> result = std::async(std::launch::async, [&] {
+    DiscIO::NANDImporter().ImportNANDBin(file.toStdString(), [&dialog, beginning] {
+      QueueOnObject(dialog, [&dialog, beginning] {
+        dialog->setLabelText(tr("Importing NAND backup\n Time elapsed: %1s")
+                                 .arg(QDateTime::currentDateTime().toSecsSinceEpoch() - beginning));
+      });
+    });
+  });
+
+  result.wait();
```

Comment:
Hmm. I think this will still block the Qt event loop. The way WiiUpdate.cpp does this is like:

```c++
auto result = std::async(std::launch::async, [&] {
  DiscIO::NANDImporter().ImportNANDBin(file.toStdString(), [&dialog, beginning] {
    QueueOnObject(dialog, [&dialog, beginning] {
      dialog->setLabelText(tr("Importing NAND backup\n Time elapsed: %1s")
                               .arg(QDateTime::currentDateTime().toSecsSinceEpoch() - beginning));
    });
  });
  QueueOnObject(dialog, [dialog] { dialog->close(); });
});

dialog->exec(); // plays nicely with the Qt event loop
result.wait(); // should be done by now, but make sure
```